modules/plugins/languages/nu.nix

```diff
@@ -0,0 +1,113 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}: let
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.types) str either package listOf enum;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.nvim.lua) expToLua;
+  inherit (lib.nvim.types) mkGrammarOption;
+  inherit (builtins) attrNames isList;
+
+  defaultServer = "nushell";
+  servers = {
+    nushell = {
+      package = pkgs.nushell;
+      lspConfig = ''
+        lspconfig.nushell.setup{
+          capabilities = capabilities,
+          on_attach = default_on_attach,
+          cmd = ${
+          if isList cfg.lsp.package
+          then expToLua cfg.lsp.package
+          else ''{"${cfg.lsp.package}/bin/nu", "--no-config-file", "--lsp"}''
+        }
+        }
+      '';
+    };
+  };
+
+  # defaultFormat = "nufmt";
+  # formats = {
+  #   nufmt = {
+  #     package = pkgs.nufmt;
+  #     nullConfig = ''
+  #       table.insert(
+  #         ls_sources,
+  #         {
+  #           name = "nufmt",
+  #           method = null_methods.internal.FORMATTING,
+  #           filetypes = { "nu" },
+  #           generator_opts = {
+  #             command = "${cfg.format.package}/bin/nufmt",
+  #             args = { "--stdin" },
+  #             to_stdin = true
+  #           },
+  #           factory = null_helpers.formatter_factory
+  #         }
+  #       )
+  #     '';
+  #   };
+  # };
```

Comment:
Is this a TODO? Or does it not work at all?

Reply:
it's a TODO, current comment version doesn't work

Reply:
TODO as "left for future work, when nufmt has a more stable schema and gets added to null-ls"